web/src/api/api.ts

```diff
@@ -75,15 +77,24 @@ export class ImmichApi {
 		this.config.basePath = baseUrl;
 	}
 
-	public getAssetFileUrl(assetId: string, isThumb?: boolean, isWeb?: boolean, key?: string) {
+	public getAssetFileUrl(
+		...[assetId, isThumb, isWeb, key]: ApiParams<typeof AssetApiFp, 'serveFile'>
```

Comment:
Can you help explain this `ApiParams<typeof AssetApiFp, 'serveFile'>`?

Reply:
The `ApiParams` type takes two parameters. The first one should match the `ApiFp` type which is basically a function that returns an object with functions (see example below). The second parameter is used to access a specific property of that object and will return the parameters of that function, except the last one.

Next, using [Labeled Tuple Elements](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-0.html#labeled-tuple-elements) the function definition appears as normal when you call it somewhere.


An example that matches `ApiFp`:

```typescript
const AssetApiFp = function(configuration) {
	return {
		async serveFile(arg0, arg1, arg2, options) {
			return;
		}
	}
}
```